### PR TITLE
telemetry: Adding telemetry for alter table reorganize partition | tidb-test=pr/2069

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Issue Number: ref #38535

Problem Summary:
Missing telemetry for new reorganize partition feature

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ type TelemetryInfo struct {
 	UseNonRecursive       bool
 	UseRecursive          bool
 	UseMultiSchemaChange  bool
-	UesExchangePartition  bool
+	UseExchangePartition  bool
```
Comment (xhebox) [inline executor/adapter.go:197]: Nice catch!

**Review by xhebox — APPROVED** [review]

File: metrics/telemetry.go [hunk 2]
```diff
@@ -269,6 +276,7 @@ type TablePartitionUsageCounter struct {
 	TablePartitionAddIntervalPartitionsCnt    int64 `json:"table_partition_add_interval_partitions_cnt"`
 	TablePartitionDropIntervalPartitionsCnt   int64 `json:"table_partition_drop_interval_partitions_cnt"`
 	TablePartitionComactCnt                   int64 `json:"table_TablePartitionComactCnt"`
```
Comment (mjonss) [inline metrics/telemetry.go:278]: (Not really part of this PR) Can this be changed to TablePartitionCompactCnt / table_TablePartitionComactCnt (adding a 'p' Comact -> Compact)? Since the json is already defined.

Comment (xhebox) [reply metrics/telemetry.go:278]: I guess It is possible. But you need to contact the telemetry(observe) team to change the column name, and change how their HTTP api handles telemetry requests.
